Update & fix TuneSDK bridge for Android

charlesTR opened the pull request:
- Bump version number of the TuneSDK bridge from 1 to 2!
- Use more recent version of TuneSDK dependency from 4.+ to 6.0.3
- Remove Java obsolete code and update to match with newer version

https://www.pivotaltracker.com/story/show/163276698

charlesTR committed:
Delete <receiver> & <service> from AndroidManifest

These 2 items would break our build because they would be
merged in the manifest at the wrong spot (outside of <application>
item).
Instead we updated our own manifest by following the native
documentation for integrating TuneSDK with Android

charlesTR committed:
Bump version code & number and play-services dependency

charlesTR committed:
Remove obsolete "*CustomProfile*" functions

We do not use those functions and they all seem deprecated

charlesTR committed:
Bump version number of TuneSDK from 4.+ to 6.0.3

Use "implementation" instead of "compile" in gradle
Remove obsolete code in the Java bridge (Tune Location)
Update Java code to match with new version of TuneSDK

fbartho commented on android/src/main/java/com/tune/react/TuneSDKBridge/TuneSDKBridge.java:
```diff
@@ -360,15 +269,33 @@ private void setUserId(Tune tune, String userIdType, String id ) {
+         TuneEventItem tuneEventItem = new TuneEventItem(eventMap.getString("itemName"));
+
+         if (eventMap.hasKey("attribute1")) {
+             tuneEventItem.withAttribute1(eventMap.getString("attribute1"));
```
Suggestion for each of the if-blocks below
```suggestion
             tuneEventItem = tuneEventItem.withAttribute1(eventMap.getString("attribute1"));
```

fbartho commented on android/build.gradle:
```diff
@@ -6,14 +6,14 @@ android {
     defaultConfig {
         minSdkVersion 16
         targetSdkVersion 22
-        versionCode 1
```
Let's undo this

charlesTR committed:
Address PR review feedback

Make sure the "builder pattern" is respected by re-assigning the
result of a mutation to its own instance:
`tuneItem = tuneItem.withAttribute(...)`

Do not bump version code/number

charlesTR committed:
Comment out the debug log lines

We could consider to delete them when we're fully done integrating
and testing TuneSDK
